core/rint/src/TTabCom.cxx

```diff
@@ -868,10 +869,17 @@ TString TTabCom::DetermineClass(const char varName[])
    type.ReadToDelim(file1, ')');
    IfDebug(std::cerr << type << std::endl);
 
-   // new version of CINT returns: "class TClassName*const)0x12345"
-   // so we have to strip off "const"
-   if (type.EndsWith("const"))
+   // new version of Cling returns: "qualifiers TClassName *const) 0x12345"
+   // we start stripping off right "const"
+   if (type.EndsWith(" *const"))
       type.Remove(type.Length() - 5);
+   // we strip off left qualifiers
+   if (type.BeginsWith("const volatile "))
+      type.Remove(0, 15);
+   else if (type.BeginsWith("const "))
+      type.Remove(0, 6);
+   else if (type.BeginsWith("volatile "))
+      type.Remove(0, 9);
```

Comment:
This can be simplified to first check for `const ` and then for `volatile `, without having to think about the combined case